models/DataObject/ClassDefinition/Data/Geopolyline.php

```diff
@@ -0,0 +1,367 @@
+
+        return null;
+    }
+
+    /**
+     * @param $importValue
+     * @param null|Model\DataObject\AbstractObject $object
+     * @param mixed $params
+     *
+     * @return array|mixed
+     */
+    public function getFromCsvImport($importValue, $object = null, $params = [])
+    {
+        $rows = explode('|', $importValue);
+        $points = [];
+        if (is_array($rows)) {
+            foreach ($rows as $row) {
+                $coords = explode(';', $row);
+                $points[] = new  DataObject\Data\Geopoint($coords[1], $coords[0]);
+            }
+        }
+
+        return $points;
+    }
+
+    /**
+     * @param $object
+     * @param mixed $params
+     *
+     * @return string
+     */
+    public function getDataForSearchIndex($object, $params = [])
+    {
+        return '';
+    }
+
+    /**
+     * converts data to be exposed via webservices
+     *
+     * @param string $object
+     * @param mixed $params
+     *
+     * @return mixed
+     */
+    public function getForWebserviceExport($object, $params = [])
+    {
+        $data = $this->getDataFromObjectParam($object, $params);
+        if (!empty($data)) {
+            return $this->getDataForEditmode($data, $object, $params);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * @param mixed $value
+     * @param null|Model\DataObject\AbstractObject $object
+     * @param mixed $params
+     * @param null $idMapper
+     *
+     * @return mixed|void
+     *
+     * @throws \Exception
+     */
+    public function getFromWebserviceImport($value, $object = null, $params = [], $idMapper = null)
+    {
+        if (empty($value)) {
+            return null;
+        } elseif (is_array($value)) {
+            $points = [];
+            foreach ($value as $point) {
+                $point = (array) $point;
+                if ($point['longitude'] != null and $point['latitude'] != null) {
+                    $points[] = new DataObject\Data\Geopoint($point['longitude'], $point['latitude']);
+                } else {
+                    throw new \Exception('cannot get values from web service import - invalid data');
+                }
+            }
+
+            return $points;
+        } else {
+            throw new \Exception('cannot get values from web service import - invalid data');
+        }
+    }
+
+    /** True if change is allowed in edit mode.
+     * @param string $object
+     * @param mixed $params
+     *
+     * @return bool
+     */
+    public function isDiffChangeAllowed($object, $params = [])
+    {
+        return true;
+    }
+
+    /** Generates a pretty version preview (similar to getVersionPreview) can be either html or
+     * a image URL. See the ObjectMerger plugin documentation for details
+     *
+     * @param $data
+     * @param null $object
+     * @param mixed $params
+     *
+     * @return array|string
+     */
+    public function getDiffVersionPreview($data, $object = null, $params = [])
```

Comment:
The docs above says it returns array or string, but the code below returns a string or void